src/LoadOrBuildDag.cpp

```diff
@@ -117,6 +118,15 @@ bool LoadOrBuildDag(Driver *self, const char *dag_fn)
         return ExitRequestingFrontendRun("%s couldn't be loaded", dag_fn);
     }
 
+    if (DetectCyclicDependencies(self->m_DagData, &self->m_Heap))
```

Comment:
You only need to do this if you get passed a new dag? `self->m_Options.m_DagFileNameJson != nullptr`

Reply:
indeed

Reply:
oh, I see what you mean.  yeah good point.